### DISCUSSION TITLE
Don't save new projects as soon as they are created

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#6677] [Ideas]
I make a lot of little tests, so I have a lot of useless, almost empty projects to be frequently removed from my account.

When the "Disable auto-save" addon is enabled, I propose that, as soon as a project is created, this is not automatically saved. This could be also made an option of this addon.

**Comment by BroJac5246** [comment]
> This could be also made an option of this addon.

I would love to see this! I also make a lot of test projects, so I typically sign out to experiment— and it's a hassle.

**Comment by DNin01** [comment]
Wouldn't the project still be "initialized" in that it would be created and a project ID would be reserved? Can we easily stop that?

An alternative would be to use SA (from source I think?) on the Scratch GUI source code hosted on GitHub Pages, at https://scratchfoundation.github.io/scratch-gui/develop. From there, you can download the projects to your computer if you need to.

I also reuse the test projects I make so things don't get out of control.

**Reply by sfederici** [reply]
> Wouldn't the project still be "initialized" in that it would be created and a project ID would be reserved? Can we easily stop that?

This is a good question

**Reply by BroJac5246** [reply]
> Wouldn't the project still be "initialized" in that it would be created and a project ID would be reserved? Can we easily stop that?
> 
> An alternative would be to use SA (from source I think?) on the Scratch GUI source code hosted on GitHub Pages, at https://scratchfoundation.github.io/scratch-gui/develop. From there, you can download the projects to your computer if you need to.
> 
> I also reuse the test projects I make so things don't get out of control.

Could we hijack the editor by loading a pre-existing project and then just uploading the new sb3 to a new project we create if the user chooses to save?

**Reply by WorldLanguages** [reply]
I believe that as soon as you go to /create, Scratch reserves a new project ID for you

**Reply by mxmou** [reply]
It sends a POST request to `projects.scratch.mit.edu` and the server responds with an ID.

**Reply by WorldLanguages** [reply]
> I believe that as soon as you go to /create, Scratch reserves a new project ID for you

Maybe I'm getting confused and this was only true in the 2.0 days.

**Comment by DNin01** [comment]
Another workaround would be to load an empty project to overwrite the contents of the tests you don’t need anymore. (Could that become an addon?)

**Reply by sfederici** [reply]
So am I going to have a lot of fully empty projects? If so, this is not what I was looking for.

**Reply by DNin01** [reply]
I meant loading an empty project file from your computer (through the File dropdown) to replace a test you no longer want to use. That way, the tests you had already created would essentially be recycled.
